src/python/grpcio/grpc/_server.py

```diff
@@ -565,8 +566,8 @@ def _send_message_callback_to_blocking_iterator_adapter(
 
 
 def _select_thread_pool_for_behavior(behavior, default_thread_pool):
-    if hasattr(behavior, 'experimental_thread_pool'
-              ) and behavior.experimental_thread_pool is not None:
+    if hasattr(behavior, 'experimental_thread_pool') and isinstance(
+            behavior.experimental_thread_pool, futures.ThreadPoolExecutor):
```

Comment:
Should we check for the base class https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor?

Reply:
I'll let you and Richard make that call :-) I hadn't really considered it, but as evidence that maybe we should not check for the base class: our server explicitly requires a `ThreadPoolExecutor` already (https://grpc.io/grpc/python/grpc.html#grpc.server) and we are actually not compatible with the other stock implementation of the base class (`ProcessPoolExecutor`, which uses fork and will break us horribly)

Reply:
I agree with Eric's point about `ProcessPoolExecutor` above. And since this is an internal API, this PR does not bind us to that decision.